monai/utils/misc.py

```diff
@@ -386,3 +391,51 @@ def sample_slices(data: NdarrayOrTensor, dim: int = 1, as_indices: bool = True,
     slices[dim] = slicevals if as_indices else slice(*slicevals)  # type: ignore
 
     return data[tuple(slices)]
+
+
+def save_obj(
+    obj, path: PathLike, create_dir: bool = True, atomic: bool = True, func: Optional[Callable] = None, **kwargs
+):
+    """
+    Save an object to file with specified path.
+    Support to serialize to a temporary file first, then move to final destination,
+    so that files are guaranteed to not be damaged if exception occurs.
+
+    Args:
+        obj: input object data to save.
+        path: target file path to save the input object.
+        create_dir: whether to create dictionary of the path if not existng, default to `True`.
+        atomic: if `True`, state is serialized to a temporary file first, then move to final destination.
+            so that files are guaranteed to not be damaged if exception occurs. default to `True`.
+        func: the function to save file, if None, default to `torch.save`.
+        kwargs: other args for the save `func` except for the checkpoint and filename.
+            default `func` is `torch.save()`, details of other args:
+            https://pytorch.org/docs/stable/generated/torch.save.html.
+
+    """
+    path = Path(path)
+    path_dir = path.parent
+    if not path_dir.exists():
+        if create_dir:
+            path_dir.mkdir(parents=True)
```

Comment:
Might have exceptions here when creating

Reply:
Hi @wyli ,

May I know which exception when `creating` you mean we should just bypass?
If the regular error, maybe we should let it raise the exception?

Thanks.